config/defaults.mk

```
@@ -169,6 +169,18 @@ MFEM_USE_BENCHMARK     = NO
 MFEM_USE_PARELAG       = NO
 MFEM_USE_ENZYME        = NO
 
+# Process MFEM_PRECISION -> MFEM_USE_SINGLE, MFEM_USE_DOUBLE
+ifneq ($(filter double Double DOUBLE,$(MFEM_PRECISION)),)
+   MFEM_USE_DOUBLE ?= YES
+   MFEM_USE_SINGLE ?= NO
+else ifneq ($(filter single Single SINGLE,$(MFEM_PRECISION)),)
+   MFEM_USE_DOUBLE ?= NO
+   MFEM_USE_SINGLE ?= YES
```

In order for this to work with a user config file, the `?=` need to be replaced with `=`:
```suggestion
   MFEM_USE_DOUBLE = YES
   MFEM_USE_SINGLE = NO
else ifneq ($(filter single Single SINGLE,$(MFEM_PRECISION)),)
   MFEM_USE_DOUBLE = NO
   MFEM_USE_SINGLE = YES
```